### PR TITLE
Mamun wip v2

### PR DESCRIPTION
**Pull request by Muazzeem**

### CODE REVIEW

File: infrastructure/importer/variables.tf [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+}
+variable "lambda_layers" {
+  default = "arn:aws:lambda:ap-southeast-1:940443069190:layer:job-post:1"
+}
+variable "job_importer_lambda_layers" {
+  default = "arn:aws:lambda:ap-southeast-1:940443069190:layer:google-map:2"
+}
+variable "environment" {
+  default = "dev"
+}
```
Comment (Muazzeem) [inline infrastructure/importer/variables.tf:35]: we are skipping the environment stage

File: infrastructure/importer/main.tf [hunk 2]
```diff
@@ -27,49 +25,104 @@ resource "aws_iam_role" "lambda_role" {
...
+  name = var.lambda_iam_policy_name
+  role = aws_iam_role.lambda_iam.id
 
-    principals {
-      type        = "Service"
-      identifiers = ["lambda.amazonaws.com"]
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
```
Comment (saifulazad) [inline infrastructure/importer/main.tf:43]: Seems like complicated. Please use AWS defined Permission

File: job_post/lambda_function.py [hunk 3]
```diff
@@ -1,66 +1,129 @@
...
-    file_name = f"{prefix}/{key}.json"
-    # print(event)
-    upload_jobs_details(
-        key=file_name, content=payload, bucket_name="extractor-service-dev"
-    )
+    if new_jobs:
+        import_results = client.collections["jobs"].documents.import_(
+            new_jobs, {"action": "upsert", "return_id": True}
+        )
+    save_job_ids(import_results)
```
Comment (Muazzeem) [inline job_post/lambda_function.py:124]: update this line inside if condition

File: infrastructure/importer/main.tf [hunk 4]
```diff
@@ -27,49 +25,104 @@ resource "aws_iam_role" "lambda_role" {
...
-  function_name = "job_importer"
-  role          = aws_iam_role.iam_for_lambda.arn
-  handler       = "lambda_function.lambda_handler"
-
+# Creating Lambda resource
+resource "aws_lambda_function" "job_post_lambda_handler" {
+  function_name    = var.function_name
+  role             = aws_iam_role.lambda_iam.arn
+  handler          = "${var.handler_name}.lambda_handler"
+  description      = "https://github.com/saifulazad/portal-scraping/tree/master/job_post"
```
Comment (saifulazad) [inline infrastructure/importer/main.tf:67]: /master/job_post?
We are at importer folder. 
So why job_post path?

File: infrastructure/importer/main.tf [hunk 5]
```diff
@@ -27,49 +25,104 @@ resource "aws_iam_role" "lambda_role" {
...
+  runtime          = var.runtime
+  timeout          = var.timeout
+  architectures    = ["arm64"]
+  layers           = [var.lambda_layers]
+  filename         = data.archive_file.archive_zip_validate.output_path
   source_code_hash = data.archive_file.archive_zip_validate.output_base64sha256
+  environment {
+    variables = {
+      API_KEY = var.typesense_api_key
```
Comment (saifulazad) [inline infrastructure/importer/main.tf:76]: TYPESENSE_API_KEY is better to use

File: infrastructure/importer/main.tf [hunk 6]
```diff
@@ -27,49 +25,104 @@ resource "aws_iam_role" "lambda_role" {
...
 
-  runtime = "python3.9"
+data "archive_file" "archive_job_importer_zip_validate" {
+  type        = "zip"
+  source_dir  = "../../job_importer"
+  output_path = "job_importer.zip"
+}
 
+# Lambda resource for import job
+resource "aws_lambda_function" "job_importer_lambda_handler" {
```
Comment (saifulazad) [inline infrastructure/importer/main.tf:88]: This lambda will be in a separate main.tf file

File: infrastructure/importer/main.tf [hunk 7]
```diff
@@ -27,49 +25,104 @@ resource "aws_iam_role" "lambda_role" {
...
   environment {
     variables = {
-      foo = "bar"
+      GOOGLE_MAP_API_KEY = var.google_map_api_key
     }
   }
 }
+
+resource "aws_lambda_function_url" "lambda_function_url" {
+  function_name      = aws_lambda_function.job_importer_lambda_handler.function_name
```
Comment (saifulazad) [inline infrastructure/importer/main.tf:105]: POST method is required

File: infrastructure/importer/variables.tf [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+variable "lambda_role_name" {
+  default = "job-post-lambda-handler-role"
+}
+
+variable "lambda_iam_policy_name" {
+  default = "job-post-lambda-handler-policy"
+}
+variable "lambda_layers" {
+  default = "arn:aws:lambda:ap-southeast-1:940443069190:layer:job-post:1"
```
Comment (saifulazad) [inline infrastructure/importer/variables.tf:28]: lambda_layers!
lambda_layer is better

File: infrastructure/importer/variables.tf [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+}
+variable "environment" {
+  default = "dev"
+}
+variable "zip_file" {
+  default = "job-post.zip"
+}
+variable "schedule" {
+  # The default value represents a cron expression scheduling a task to run every day at 6:20 PM. https://crontab.cronhub.io
+  default = "cron(20 18 * * ? *)"
```
Comment (saifulazad) [inline infrastructure/importer/variables.tf:41]: Why at 6:20 PM?

File: job_post/settings.py [hunk 10]
```diff
@@ -0,0 +1 @@
+BUCKET_NAME = "extractor-service-dev"
```
Comment (saifulazad) [inline job_post/settings.py:1]: Keep in same file. Not other file.

**Review by saifulazad — CHANGES_REQUESTED** [review]

**Pull request merged**